stack.yaml

```diff
@@ -131,3 +142,11 @@ ghc-options:
   cardano-sl:            -Werror
   cardano-sl-tools:      -Werror
   cardano-sl-lwallet:    -Werror
+  cardano-sl-explorer:   -Werror
+
+# Added for explorer
+flags:
+  cardano-sl:
+    with-explorer: True
```

Comment:
> AFAIU, if we want Stack to compile explorer, which is something that we do want - we have a single stack.yaml and need to compile all the dependencies in one go.

Yes. Once again, if you want to change a default value of `with-explorer`, it should be done in `cardano-sl.cabal`, because it's simpler (otherwise it's not obvious which one will be used by default: the one from cabal file or the one from stack.yaml). That's the first point.
The second point is that I don't like changing the default value. The build without explorer should be used more often (we need to build explorer less often than cluster nodes or wallets). If one wants to build explorer and doesn't enable `with-explorer`, the build will fail which is good because we will know about misconfiguration. However, if one wants to build a regular node and enables `with-explorer` the build will pass, but the node will have explorer part (which is not critical, but is still bad, because it adds some overhead and extra possibility of a bug). Also I don't think it's a good idea to change the default behavior unless there is a very good reason for it (currently I don't see one).

> But where would that script be? 

There are two scripts: one for developers and one for CI. For developers I think you should just modify `scripts/build/cardano-sl.sh` as I already said. Regarding CI, I am not sure what our current plan is, but in the simplest case you can just pass `with-explorer` flag from there.